### PR TITLE
1430/listings management, fcfs and lottery

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue

Addresses #1430 
- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds the lottery and first come first serve radio button selection to the rankings and results portion of listings management.

If you select FCFS, an additional radio question `Is there an application due date?` will appear - if you select `No`, the application due date field at the bottom of the page should become disabled

If you select Lottery, three additional fields for the date, start time, and end time, as well as a textbox field for notes should appear

I added a computed backend field to be able to tell if a listing is FCFS.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Add a new listing with FCFS and no due date, and check that the public listing displays "First come first serve"
Add a new listing with Lottery and check out the additional lottery section with the time range and notes in the right hand panel

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes

### CODE REVIEW

**Comment by emilyjablonski** [comment]
Large prettier diff is making it hard to review, taking a look at that

File: backend/core/src/listings/entities/listing.entity.ts [hunk 1]
```diff
@@ -410,6 +412,19 @@ class Listing extends BaseEntity {
...
+  get reviewOrderType(): ListingReviewOrder {
+    let lotteryEventExists = false
+    this.events.forEach((event) => {
+      if (event.type === ListingEventType.publicLottery) {
+        lotteryEventExists = true
+      }
+    })
+    return lotteryEventExists ? ListingReviewOrder.lottery : ListingReviewOrder.firstComeFirstServe
+  }
+
```
Comment (seanmalbert) [inline backend/core/src/listings/entities/listing.entity.ts:427]: I think this should be moved to the listing DTO. See the other definitions in there with the `@Transform` decorator. Also, with the way it is now generate client doesn't properly add `reviewOrderType` or `ListingReviewOrder`, which causes issues, so the definition should include `reviewOrderType: ListingReviewOrder`.

**Review by seanmalbert — CHANGES_REQUESTED** [review]
@emilyjablonski ,
Overall this looks good. The main change that's needed is how reviewOrderType is defined in the backend. Please see my comment on that. 

My machine is probably a little slow at the moment, but it does seem to take a long time to switch between the first come first serve and lottery options in the form. If it seems a little slow to you too, consider moving those pieces to a memoized component(s).

**Comment by pbn4** [comment]
@emilyjablonski Could you please try to revert the formatting commits? I can make it a separate PR.

**Pull request closed without merging**